### PR TITLE
Fixing SCRIPT_NAME and PATH_INFO for non-root applications

### PR DESCRIPTION
**Pull request by adamkerz**
Django depends on SCRIPT_NAME and PATH_INFO adhering to PEP333 (http://www.python.org/dev/peps/pep-0333/)

See issue: #1029

### CODE REVIEW

**Comment by crwilcox** [comment]
:+1:

File: Python/Product/WFastCgi/wfastcgi.py [hunk 1]
```diff
@@ -803,6 +803,13 @@ def main():
                     record.params['SCRIPT_NAME'] = ''
                     record.params['wsgi.script_name'] = wsgi_encode('')
 
+                # correct SCRIPT_NAME and PATH_INFO if we are told what our SCRIPT_NAME should be
+                if 'SCRIPT_NAME' in os.environ:
```
Comment (zooba) [inline Python/Product/WFastCgi/wfastcgi.py]: For safety, I'd prefer to check that `PATH_INFO` starts with `SCRIPT_NAME` as part of this condition.

**Comment by zooba** [comment]
The [old IIS docs](https://msdn.microsoft.com/en-us/library/ms524602(v=vs.90).aspx) seem to suggest that you can set some properties to get the right behaviour within the server. I can't easily confirm right now, and I can't find any more recent references, but maybe this is going to be a better solution? (It may be very .NET centric though... so might not actually help.)

**Comment by adamkerz** [comment]
How about that check - if PATH_INFO starts with SCRIPT_NAME (case insensitive)?

Also, `AllowPathInfoForScriptMappings` was replaced with `allowPathInfo` but when I tried that, it didn't do what it says. The [v7.0 docs](https://msdn.microsoft.com/en-us/library/ms691481%28v=vs.90%29.aspx) say that it's set to do the right thing by default (but it doesn't) and when I set it to true, I think it just cleared SCRIPT_NAME and left PATH_INFO as the full path. (It was a week or two ago that I tried it out). Either way, when setting it to true or false, neither produced the correct SCRIPT_NAME+PATH_INFO behaviour.

**Comment by adamkerz** [comment]
If you figure out how to get IIS to do the right thing on it's own, I'm more than happy to test and use that instead of this slight hack!

**Comment by zooba** [comment]
I'm happy with this. Can't guarantee when we'll make a release for wfastcgi (I'm keen to split it out and rewrite the tests in Python), but it sounds like you're unblocked anyway and there's no rush.

**Comment by adamkerz** [comment]
Correct, I am unblocked so no rush. Thanks for your help on this!

**Pull request merged**